core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.spec.ts

```diff
@@ -0,0 +1,223 @@
+    subtopic1.addSkill('skill1', 'subtopic1 skill');
+    topic.getSubtopics = function() {
+      return [subtopic1];
+    };
+
+    spyOn(TopicEditorStateService, 'getTopic').and.returnValue(topic);
+    spyOn(focusManagerService, 'setFocus');
+    spyOnProperty(TopicEditorStateService, 'onTopicInitialized').and.callFake(
+      function() {
+        return topicInitializedEventEmitter;
+      });
+    spyOnProperty(
+      TopicEditorStateService, 'onTopicReinitialized').and.callFake(
+      function() {
+        return topicReinitializedEventEmitter;
+      });
+    ctrl.$onInit();
+  }));
+
+  afterEach(() => {
+    ctrl.$onDestroy();
+  });
+
+  it('should initialize the variables', function() {
+    const topicRights = TopicRights.createInterstitialRights();
+    const allSkillSummaries = subtopic1.getSkillSummaries();
+    spyOn(TopicEditorStateService, 'getSkillIdToRubricsObject').and
+      .returnValue(skillIdToRubricsObject);
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+    expect($scope.selectedSkillId).toBeNull();
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+    expect($scope.topicRights).toEqual(topicRights);
+    expect($scope.skillIdToRubricsObject).toEqual(skillIdToRubricsObject);
+    expect($scope.allSkillSummaries).toEqual(allSkillSummaries);
+    expect($scope.getSkillsCategorizedByTopics).toBe(CategorizedSkillsDict);
+    expect($scope.getUntriagedSkillSummaries)
+      .toBe(UntriagedSkillSummaries);
+    expect($scope.canEditQuestion).toBe(false);
+    expect($scope.misconceptions).toEqual([]);
+    expect($scope.questionIsBeingUpdated).toBe(false);
+    expect($scope.questionIsBeingSaved).toBe(false);
+    expect($scope.emptyMisconceptionsList).toEqual([]);
+  });
+
+  it('should setFocus on selectSkillField when screen loads', function() {
+    $window.onload();
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+  });
+
+  it('should call initTab when topic is initialized', function() {
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+
+    topicInitializedEventEmitter.emit();
+
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+  });
+
+  it('should call initTab when topic is reinitialized', function() {
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+
+    topicInitializedEventEmitter.emit();
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+    topicReinitializedEventEmitter.emit();
+
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+  });
+
+  it('should unsubscribe ondestroy', function() {
```

Comment:
Is this description correct?

Reply:
I changed the description to `should unsubscribe when onDestroy runs`. DONE